tests/unit/test_blob.py

```diff
@@ -1785,6 +1897,10 @@ def _upload_from_file_helper(self, side_effect=None, **kwargs):
         content_type = u"font/woff"
         client = mock.sentinel.client
         predefined_acl = kwargs.get("predefined_acl", None)
+        generation_match = kwargs.get("generation_match", None)
```

Comment:
One more request, could you add system tests that exercise these new parameters?